### ISSUE TITLE
Rolling video on CRT when using YPbPr & IO Board, until a white screen.

### ISSUE THREAD

**Issue by jrronimo** [MiSTer-devel/Genesis_MiSTer#150]
I've been trying to track this down for a few days. I'm not sure if this is the right place, but it's a pretty repeatable issue.

I have my MiSTer equipped with an IO Board v6.1. I'm using VGA to RGB cables and plugging into the component inputs on a Philips 27PT543S37A CRT. I'm using ypbpr=1 in MiSTer.ini. The Genesis core doesn't sync (for lack of a better term) at the OSD. If you can load a game anyway, the game also doesn't sync, unless it gets to a really bright/white screen. For instance, any Konami game will sync fine when the Konami logo fills the screen with white.

Here's a video showing this in action just after selecting Zombies Ate My Neighbors: [https://youtu.be/dBu-MtANCQI?t=187](https://youtu.be/dBu-MtANCQI?t=187)

Once the video has sync'd, you can pull up the OSD fine. If you load a different game, it will be out of sync again until there's a bright screen. Streets of Rage 1, for instance, is unplayable until WAY into the intro when there's a white flash, then the game is fine.

I've tried to set many different video_modes, both in the main section of the MiSTer.ini file and in a sub [Genesis] section. I've tried setting the video_mode to be the same as Genesis games, but that doesn't seem to have any effect.

A regular Genesis displays video on this TV just fine, though I don't have one that's RGB modded.

I know this is a little outside the scope of the MiSTer project and that HDMI is the preferred, but there have been a few other reports of this issue (see: [https://misterfpga.org/viewtopic.php?p=13121#p13121](https://misterfpga.org/viewtopic.php?p=13121#p13121) ).

**I think there can be an easy fix, too:** The NES core shows a bright grey or pink screen when it fires up, and has a bright background for the OSD, and it syncs every time. Every NES game seems to load fine and not have any sync issues with this same exact setup. Would it be possible to have a toggle or a .ini setting that displays the OSD on a white background when the Genesis core starts up? It can be a picture or just a white filled screen.

I'm not sure if that would fix the issue once a game has been loaded though... but I would be really happy to test this. I started looking at the MiSTer source code to see if I could figure out how to add a white screen when the OSD loads, or maybe when pushing the USER button, or anything I can think of, but it's definitely over my head.

Thanks for your time and all your hard work. I really appreciate what goes into this project!

**Comment by sorgelig** [comment]
You didn't mention if set SOG switch to ON position.

**Comment by jrronimo** [comment]
> You didn't mention if you set SOG switch to ON position.

Whoops, my mistake. It is currently set to ON (it is in the OVR position). If I set it to Auto, the Menu core rolls as well.

**Comment by sorgelig** [comment]
May be sync level is not enough for your TV. Component output on MiSTer is simplified and kind of hacky.
You may also try both composite_sync=0 and composite_sync=1 in ini.

**Comment by jrronimo** [comment]
I understand. I tried with both composite_sync=0 and composite_sync=1 and it didn't change. 

Would it be feasible to have an option for a white background for the OSD when the core first launches? Or is there a way to boost the sync level?

Thanks for looking at this.

**Comment by sorgelig** [comment]
It's HW issue. You need a better converter from RGB to Component. Or at least some sync level boost. It can't be fixed by software.
Btw, make sure your cable has no additional circuit inside which may decrease the sync level.

**Comment by jrronimo** [comment]
I'll check my cable for sure.

I was hoping that there could be a solution since the NES core seems to work well, but I'll look for a better converter. In the meantime, I made a very simple Genesis ROM to just display a colored screen which I've set as boot0.rom. It makes the Menu core sync every time at the moment. I can at least play some games. :)

One last thing I noticed, just in case it matters:
In the sys_top.v file for the Genesis core, it appears that the VGA Scaler is being forced on (line 290 - "wire       vga_fb       = cfg[12] | vga_force_scaler;"), but for the NES core this is not the case ("wire       vga_fb       = cfg[12];"). SNES also appears to have it forced, and I'm seeing the rolling picture there, too.

I've noticed that if I enable vga_scaler=1 in MiSTer.ini, then the Menu core has a similarly rolling picture over my component cables. Is the VGA scaler actually being forced for Genesis/SNES? Is it possible to turn it off to see if that helps my problem?

If the VGA Scaler needs to be on, I understand. I just wanted to ask before I close to satisfy my curiosity.

Thanks again for your insight!
